make build err,Why?

make build
docker build \
        --build-arg uid=0 \
        -t quay.io/spotahome/redis-operator-dev:latest \
        -t quay.io/spotahome/redis-operator-dev:632aa3da88ee2a28ccb9fc4f571ddcb556dbd713 \
        -f docker/development/Dockerfile \
        .
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            Install the buildx component to build images with BuildKit:
            https://docs.docker.com/go/buildx/

Sending build context to Docker daemon  61.99MB
Step 1/11 : FROM golang:1.20-alpine
 ---> 71719a2da3d1
Step 2/11 : ENV CODEGEN_VERSION="1.11.9"
 ---> Using cache
 ---> e86d68844aa2
Step 3/11 : RUN apk --no-cache add     bash     git     g++     openssl
 ---> Using cache
 ---> ce171fba21d4
Step 4/11 : RUN wget http://github.com/kubernetes/code-generator/archive/kubernetes-${CODEGEN_VERSION}.tar.gz &&     mkdir -p /go/src/k8s.io/code-generator/ &&     tar zxvf kubernetes-${CODEGEN_VERSION}.tar.gz --strip 1 -C /go/src/k8s.io/code-generator/ &&     mkdir -p /go/src/k8s.io/kubernetes/hack/boilerplate/ &&     touch /go/src/k8s.io/kubernetes/hack/boilerplate/boilerplate.go.txt
 ---> Using cache
 ---> fdaea51f620d
Step 5/11 : ARG MOCKERY_VERSION="2.9.6"
 ---> Using cache
 ---> e2b3b3e790c1
Step 6/11 : RUN wget -c https://github.com/vektra/mockery/releases/download/v${MOCKERY_VERSION}/mockery_${MOCKERY_VERSION}_$(uname -o)_$(uname -m).tar.gz -O - | tar -xz -C /go/bin/
 ---> Using cache
 ---> ddb48897503d
Step 7/11 : ARG uid=1000
 ---> Using cache
 ---> e7f525a1f6cd
Step 8/11 : ARG gid=1000
 ---> Using cache
 ---> 00465870281f
Step 9/11 : RUN addgroup -g $gid rf &&     adduser -D -u $uid -G rf rf &&     chown rf:rf -R /go
 ---> Running in 1ae9b85a2b77
adduser: uid '0' in use
The command '/bin/sh -c addgroup -g $gid rf &&     adduser -D -u $uid -G rf rf &&     chown rf:rf -R /go' returned a non-zero code: 1
make: *** [Makefile:68: docker-build] Error 1

--- a/docker/development/Dockerfile
+++ b/docker/development/Dockerfile
@@ -23,10 +23,11 @@ RUN wget -c https://github.com/vektra/mockery/releases/download/v${MOCKERY_VERSI
 # Create user
 ARG uid=1000
 ARG gid=1000
-RUN addgroup -g $gid rf && \
-    adduser -D -u $uid -G rf rf && \
-    chown rf:rf -R /go
+RUN if ! getent passwd "$uid" > /dev/null 2>&1; then \
+      addgroup -g "$gid" rf && \
+      adduser -D -u "$uid" -G rf rf; \
+    fi && \
+    chown -R "$uid":"$gid" /go
 
-
-USER rf
+USER $uid
 WORKDIR /go/src/github.com/saremox/redis-operator

